dali/kernels/signal/dct/dct_gpu.cu

```diff
@@ -0,0 +1,185 @@
+        for (int i = 0; i < sample.input_length; ++i) {
+          out_val += *input * cos_row[i];
+          input += in_stride[1];
+        }
+        sample.output[output_idx] = out_val;
+      }
+    }
+  }
+}
+
+template <typename OutputType, typename InputType, int Dims>
+KernelRequirements Dct1DGpu<OutputType, InputType, Dims>::Setup(KernelContext &ctx,
+                                                               const InListGPU<InputType, Dims> &in,
+                                                               span<const DctArgs> args,
+                                                               int axis) {
+  DALI_ENFORCE(args.size() == in.num_samples());
+  KernelRequirements req{};
+  ScratchpadEstimator se{};
+  args_.clear();
+  cos_tables_.clear();
+  sample_descs_.clear();
+  TensorListShape<Dims> out_shape(in.num_samples());
+  TensorListShape<3> reduced_shape(in.num_samples());
+  max_cos_table_size_ = 0;
+  axis_ = axis >= 0 ? axis : Dims - 1;
+  DALI_ENFORCE(axis_ >= 0 && axis_ < Dims,
+               make_string("Axis is out of bounds: ", axis_));
+  for (int s = 0; s < args.size(); ++s) {
+    args_.push_back(args[s]);
+    auto &arg = args_.back();
+    int64_t n = in.shape[s][axis_];
+
+    if (arg.dct_type == 1) {
+      DALI_ENFORCE(n > 1, "DCT type I requires an input length > 1");
+      if (arg.normalize) {
+        DALI_WARN("DCT type-I does not support orthogonal normalization. Ignoring");
+        arg.normalize = false;
+      }
+    }
+
+    if (arg.ndct <= 0) {
+      arg.ndct = n;
+    }
+    if (cos_tables_.find({n, arg}) == cos_tables_.end()) {
+      cos_tables_[{n, arg}] = nullptr;
+      se.add<OutputType>(AllocType::GPU, n * arg.ndct);
+      if (n * arg.ndct > max_cos_table_size_) {
+        max_cos_table_size_ = n * arg.ndct;
+      }
+    }
+    auto in_shape = in.tensor_shape_span(s);
+    reduced_shape.set_tensor_shape(s, reduce_shape(in_shape, axis_, arg.ndct));
+    auto sample_shape = in.shape[s];
+    sample_shape[axis_] = arg.ndct;
+    out_shape.set_tensor_shape(s, sample_shape);
+  }
+  se.add<OutputType>(AllocType::Pinned, max_cos_table_size_);
+  if (cos_tables_.size() > 1) {
+    se.add<OutputType>(AllocType::Pinned, max_cos_table_size_);
+  }
+  se.add<SampleDesc>(AllocType::GPU, in.num_samples());
+  block_setup_.SetupBlocks(reduced_shape, true);
+  se.add<BlockDesc<3>>(AllocType::GPU, block_setup_.Blocks().size());
+  req.output_shapes = {out_shape};
+  req.scratch_sizes = se.sizes;
+  return req;
+}
+
+template <typename OutputType,  typename InputType, int Dims>
+DLL_PUBLIC void Dct1DGpu<OutputType, InputType, Dims>::Run(KernelContext &ctx,
+                                                          const OutListGPU<OutputType, Dims> &out,
+                                                          const InListGPU<InputType, Dims> &in,
+                                                          span<const DctArgs> args, int axis) {
+  (void)args;
+  (void)axis;
```

Comment:
I think you can just comment out the arg names from the signature definition.

Reply:
Good point, done